apidef/api_definitions.go

```diff
@@ -551,6 +553,62 @@ type GraphQLConfig struct {
 	TypeFieldConfigurations []datasource.TypeFieldConfiguration `bson:"type_field_configurations" json:"type_field_configurations"`
 	// GraphQLPlayground is the Playground specific configuration.
 	GraphQLPlayground GraphQLPlayground `bson:"playground" json:"playground"`
+	// EngineConfig holds the configuration for engine v2 and upwards
+	EngineConfig GraphQLEngineConfig `bson:"engine_config" json:"engine_config"`
+}
+
+type GraphQLConfigVersion string
+
+const (
+	GraphQLConfigVersionNone = ""
+	GraphQLConfigVersion1    = "1"
+	GraphQLConfigVersion2    = "2"
+)
+
+type GraphQLEngineConfig struct {
+	FieldConfigs []GraphQLFieldConfig      `bson:"field_configs" json:"field_configs"`
+	DataSources  []GraphQLEngineDataSource `bson:"data_sources" json:"data_sources"`
+}
+
+type GraphQLFieldConfig struct {
+	TypeName              string   `bson:"type_name" json:"type_name"`
+	FieldName             string   `bson:"field_name" json:"field_name"`
+	DisableDefaultMapping bool     `bson:"disable_default_mapping" json:"disable_default_mapping"`
+	Path                  []string `bson:"path" json:"path"`
+}
+
+type GraphQLEngineDataSourceKind string
+
+const (
+	GraphQLEngineDataSourceKindREST    = "REST"
+	GraphQLEngineDataSourceKindGraphQL = "GraphQL"
+)
+
+type GraphQLEngineDataSource struct {
+	Kind       GraphQLEngineDataSourceKind `bson:"kind" json:"kind"`
+	Name       string                      `bson:"name" json:"name"`
+	Internal   bool                        `bson:"internal" json:"internal"`
+	RootFields []GraphQLTypeFields         `bson:"root_fields" json:"root_fields"`
+	Config     json.RawMessage             `bson:"config" json:"config"`
+}
+
+type GraphQLTypeFields struct {
```

Comment:
The library already has a graphql package and putting apidef related things in another graphql package would be very confusing - besides the fact they are apidef-related (they should really live in this package)

Reply:
They will be inside `apidef` but in a separate package. There are very long names all starting with `GraphQL`, it is a little awkward.

Reply:
all other apidef related types are located in this package - it would be unexpected to have the GraphQL ones in another. Also, long names are not bad even though the Go Community don't like them